momento-sdk/src/main/java/momento/sdk/ScsTokenClient.java

```diff
@@ -0,0 +1,217 @@
+package momento.sdk;
+
+import com.google.gson.Gson;
+import com.google.protobuf.ByteString;
+import grpc.permission_messages.CacheRole;
+import grpc.permission_messages.ExplicitPermissions;
+import grpc.permission_messages.Permissions;
+import grpc.permission_messages.PermissionsType;
+import grpc.permission_messages.TopicRole;
+import java.util.concurrent.CompletableFuture;
+import javax.annotation.Nonnull;
+import momento.sdk.auth.CredentialProvider;
+import momento.sdk.auth.accessControl.CacheItemSelector;
+import momento.sdk.auth.accessControl.CacheSelector;
+import momento.sdk.auth.accessControl.DisposableToken;
+import momento.sdk.auth.accessControl.DisposableTokenPermission;
+import momento.sdk.auth.accessControl.DisposableTokenScope;
+import momento.sdk.auth.accessControl.ExpiresIn;
+import momento.sdk.auth.accessControl.TopicSelector;
+import momento.sdk.exceptions.InvalidArgumentException;
+import momento.sdk.exceptions.UnknownException;
+import momento.sdk.responses.auth.GenerateDisposableTokenResponse;
+import momento.token._GenerateDisposableTokenRequest;
+import momento.token._GenerateDisposableTokenResponse;
+
+public class ScsTokenClient {
```

Comment:
you don't want this class to be `public`, otherwise it can be referenced directly by users and will show up in IDE hints. Check out the other `Scs*` classes for examples.